docs/developer/data_address_types.md

```diff
@@ -0,0 +1,124 @@
+# A Note On Data Address Types
+
+A **_Data Address_** is a location reference used by EDC runtimes. This note outlines the different data address types and their use. 
```

Comment:
one question from a discussion we have in parallel:
Is the data address just the **location** of the data or does it also include **"all knowledge to retrieve the data"**. E.g. in case of an HTTP typed address: is it just the URL or all values that are needed to perform an HTTP call to retrieve the data (potentially including http method, http headers or even a body that needs to be passed).

Reply:
Thanks, I agree, especially with the out-of-band storage of secrets. 
I thought about edge cases. What if data can just be retrieved if a special http header is set, for example `Accept` to define the expected mime type? 
That cannot be included in a URL. So it's no just a "location reference" but also any knowledge that is needed to retrieve data.